gwvolman/tasks.py

```diff
@@ -708,3 +653,173 @@ def rebuild_image_cache():
             logging.info("Build time: %i seconds", elapsed)
 
         shutil.rmtree(temp_dir, ignore_errors=True)
+
+def _mount_girderfs(mountpoint, directory, fs_type, obj_id, api_key):
+    """Mount a girderfs directory given a type and id"""
+    cmd = (
+        f"girderfs -c {fs_type} --api-url {GIRDER_API_URL} --api-key {api_key}"
+        f" {os.path.join(mountpoint, directory)} {obj_id}"
+    )
+    logging.info("Calling: %s", cmd)
+    subprocess.call(cmd, shell=True)
+    print(f"Mounted {fs_type} {directory}")
+
+def _make_fuse_dirs(mountpoint, directories):
+    """Create fuse directories""" 
+
+    # FUSE is silly and needs to have mirror inside container
+    for suffix in directories:
+        directory = os.path.join(mountpoint, suffix)
+        _safe_mkdir(HOSTDIR + directory)
+        if not os.path.isdir(directory):
+            os.makedirs(directory)
+
+def _create_docker_volume(cli, vol_name):
+    """Creates a Docker volume with the specified name"""
+
+    try:
+        volume = cli.volumes.create(name=vol_name, driver='local')
+    except DockerException as dex:
+        logging.error(f"Error creating volume {vol_name} using local driver")
+        logging.exception(dex)
+
+    logging.info(f"Volume: {volume.name} created")
+    mountpoint = volume.attrs['Mountpoint']
+    logging.info(f"Mountpoint: {mountpoint}")
+
+    os.chown(HOSTDIR + mountpoint, DEFAULT_USER, DEFAULT_GROUP)
+    for root, dirs, files in os.walk(HOSTDIR + mountpoint):
+        for obj in dirs + files:
+            os.chown(os.path.join(root, obj), DEFAULT_USER, DEFAULT_GROUP)
+
+    # Before calling girderfs and "escaping" container, we need to make
+    # sure that shared objects we use are available on the host
+    # TODO: this assumes overlayfs
+    mounts = ''.join(open(HOSTDIR + '/proc/1/mounts').readlines())
+    if 'overlay /usr/local' not in mounts:
+        cont = cli.containers.get(socket.gethostname())
+        libdir = cont.attrs['GraphDriver']['Data']['MergedDir']
+        subprocess.call('mount --bind {}/usr/local /usr/local'.format(libdir),
+                        shell=True)
+    return mountpoint
+
+def _get_session(gc, tale=None, version_id=None):
+    """Returns the session for a tale or version"""
+    session = {'_id': None}
+
+    if tale is not None and tale.get('dataSet') is not None:
+        session = gc.post(
+            '/dm/session', parameters={'taleId': tale['_id']})
+    elif version_id is not None:
+        # Get the dataset for the version
+        dataset = gc.get('/version/{}/dataSet'.format(version_id))
+        if dataset is not None:
+            session = gc.post(
+                '/dm/session', parameters={'dataSet': json.dumps(dataset)})
+
+    return session
+
+
+@girder_job(title='Recorded Run')
+@app.task(bind=True)
+def recorded_run(self, run_id, tale_id, user_id):
```

Comment:
This should be authenticated request. I don't think passing `user_id` is necessary